home/models.py

```diff
@@ -100,6 +101,24 @@ class Meta:
     def __str__(self):
         return f"Question #{self.id} : {self.title} ( {self.publish_date.strftime('%d/%m/%Y %H:%M')} )"
 
+    def create_answers_structure(self, filter_type, current_profile):
```

Comment:
Having, looked at this again - I now understand this is a querying function that returns a list of answers. The function name is a bit misleading beucause I usually interpret "structure" to mean something like "struct" in C. 

Also this being an object-level method, one expects it to return object-level data and not a full list of objects.

It seems like the purpose of this is to get icons for display in the UI. The same could be achieved by adding a couple of property getters to the answers object instead of a whole wrapper to the `get_answers_feed` class method.

But this is mixing UI concerns such as icon name with model code - I would advise to keep the UI-specific stuff in the UI code. The model can still have helper functions to make it easy to know if an answer is liked or not, but things like CSS class names belong in the UI layer.

Reply:
I changed it as you suggested. Now having a tuple array of:
(answer object, bool is_user_liked, bool is_user_disliked)
I use the bools to pick which icon to display in the template